xsd-core/src/main/java/io/ballerina/xsdtorecordconverter/XSDToRecord.java

```diff
@@ -0,0 +1,181 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.xsdtorecordconverter;
+
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.NodeParser;
+import io.ballerina.xsdtorecordconverter.component.XSDComponent;
+import io.ballerina.xsdtorecordconverter.visitor.VisitorUtils;
+import io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Locale;
+import java.util.Objects;
+
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.CLOSE_BRACES;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.COMMA;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.CONTENT_FIELD;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.ENUM;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.NAME;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.OPEN_BRACES;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.PUBLIC;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.SEMICOLON;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.VERTICAL_BAR;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.WHITESPACE;
+
+/**
+ * This class is used for transforming an XSD into a corresponding record format.
+ */
+public final class XSDToRecord {
+    public static final String SCHEMA = "schema";
+    public static final String EOF_TOKEN = "";
+    public static final String INVALID_IMPORTS_ERROR = "Invalid imports have been found.";
+    public static final String INVALID_XSD_FORMAT_ERROR = "The provided XML document is not a valid XSD schema. " +
+            "The root element must be a <schema>.";
+    public static final String XMLDATA_NAME_ANNOTATION = "@xmldata:Name {value: \"%s\"}";
+
+    public static String convert(Document document) throws Exception {
+        Element rootElement = document.getDocumentElement();
+        if (!Objects.equals(rootElement.getLocalName(), SCHEMA)) {
+            throw new Exception(INVALID_XSD_FORMAT_ERROR);
+        }
+        XSDVisitorImpl xsdVisitor = new XSDVisitorImpl();
+        HashMap<String, ModuleMemberDeclarationNode> nodes = new LinkedHashMap<>();
+        processNodeList(rootElement, nodes, xsdVisitor);
+        ModulePartNode modulePartNode = Utils.generateModulePartNode(nodes, xsdVisitor);
+        return Utils.formatModuleParts(modulePartNode);
+    }
+
+    private static void processNodeList(Element rootElement, HashMap<String, ModuleMemberDeclarationNode> nodes,
```

Comment:
Same as this comment [1]

[1] - https://github.com/ballerina-platform/xsd-tools/pull/1/files#r1879305050